### PR TITLE
[doc] Update DeepSpeech 

### PR DESCRIPTION
**Pull request by pavel-esir**
### Details:
 - cherry picked from #https://github.com/openvinotoolkit/openvino/pull/6011
### Tickets:
 - 54733, 54739, 38915

### CODE REVIEW

File: docs/MO_DG/prepare_model/convert_model/tf_specific/Convert_DeepSpeech_From_Tensorflow.md [hunk 1]
```diff
@@ -2,66 +2,80 @@
...
 
-It helps the model to remember the context of the words that it takes as input.
+At the first inference the variables are initialized by zero tensors. After executing, the results of the `BlockLSTM` 
+are assigned to cell state and hidden state, which are these two variables.
 
-## Convert the TensorFlow* DeepSpeech Model to IR
+## Convert the Main Part of DeepSpeech Model into IR
 
-The Model Optimizer assumes that the output model is for inference only. That is why you should cut those variables off and resolve keeping cell and hidden states on the application level.
+The Model Optimizer assumes that the output model is for inference only. That is why you should cut `previous_state_c`
```
Comment (lazarevevgeny) [inline docs/MO_DG/prepare_model/convert_model/tf_specific/Convert_DeepSpeech_From_Tensorflow.md:61]: Do not use "The" in the beginning of the sentence.

File: docs/MO_DG/prepare_model/convert_model/tf_specific/Convert_DeepSpeech_From_Tensorflow.md [hunk 2]
```diff
@@ -2,66 +2,80 @@
...
-The Model Optimizer assumes that the output model is for inference only. That is why you should cut those variables off and resolve keeping cell and hidden states on the application level.
+The Model Optimizer assumes that the output model is for inference only. That is why you should cut `previous_state_c` 
+and `previous_state_h` variables off and resolve keeping cell and hidden states on the application level.
 
 There are certain limitations for the model conversion:
 - Time length (`time_len`) and sequence length (`seq_len`) are equal.
 - Original model cannot be reshaped, so you should keep original shapes.
 
-To generate the DeepSpeech Intermediate Representation (IR), provide the TensorFlow DeepSpeech model to the Model Optimizer with the following parameters:
+To generate the IR, run Model Optimizer with the following parameters:
```
Comment (lazarevevgeny) [inline docs/MO_DG/prepare_model/convert_model/tf_specific/Convert_DeepSpeech_From_Tensorflow.md:68]: run the Model Optimizer

File: docs/MO_DG/prepare_model/convert_model/tf_specific/Convert_DeepSpeech_From_Tensorflow.md [hunk 3]
```diff
@@ -2,66 +2,80 @@
...
+--output "cudnn_lstm/rnn/multi_rnn_cell/cell_0/cudnn_compatible_lstm_cell/GatherNd_1,cudnn_lstm/rnn/multi_rnn_cell/cell_0/cudnn_compatible_lstm_cell/GatherNd,logits" \
 --disable_nhwc_to_nchw
 ```
 
 Where:
-* `--freeze_placeholder_with_value input_lengths->[16]` freezes sequence length
-* `--input input_node,previous_state_h/read,previous_state_c/read` and
-`--input_shape [1,16,19,26],[1,2048],[1,2048]` replace the variables with a placeholder
-* `--output raw_logits,lstm_fused_cell/GatherNd,lstm_fused_cell/GatherNd_1` gets data for the next model
+* `input_lengths->[16]` freezes sequence length
```
Comment (lazarevevgeny) [inline docs/MO_DG/prepare_model/convert_model/tf_specific/Convert_DeepSpeech_From_Tensorflow.md:78]: Please, see the comment from the master branch.

**Review by lazarevevgeny — CHANGES_REQUESTED** [review]

File: docs/MO_DG/prepare_model/convert_model/tf_specific/Convert_DeepSpeech_From_Tensorflow.md (hunk 3 above)
Comment (pavel-esir) [reply docs/MO_DG/prepare_model/convert_model/tf_specific/Convert_DeepSpeech_From_Tensorflow.md]: For some reason my cherry picked commit was not pushed properly. Pushed again with corrections

File: docs/MO_DG/prepare_model/convert_model/tf_specific/Convert_DeepSpeech_From_Tensorflow.md (hunk 2 above)
Comment (pavel-esir) [reply docs/MO_DG/prepare_model/convert_model/tf_specific/Convert_DeepSpeech_From_Tensorflow.md]: fixed

File: docs/MO_DG/prepare_model/convert_model/tf_specific/Convert_DeepSpeech_From_Tensorflow.md (hunk 3 above)
Comment (pavel-esir) [reply docs/MO_DG/prepare_model/convert_model/tf_specific/Convert_DeepSpeech_From_Tensorflow.md]: For some reason my cherry picked commit was not pushed properly. Pushed again with corrections

File: docs/MO_DG/prepare_model/convert_model/tf_specific/Convert_DeepSpeech_From_Tensorflow.md (hunk 1 above)
Comment (pavel-esir) [reply docs/MO_DG/prepare_model/convert_model/tf_specific/Convert_DeepSpeech_From_Tensorflow.md]: removed

**Review by lazarevevgeny — APPROVED** [review]

**Pull request merged**